docs.feldera.com/docs/pipelines/transactions.md

```diff
@@ -0,0 +1,256 @@
+    ```shell
+    $ fda start-transaction my_pipeline
+
+    Transaction started successfully with ID: 1
+    ```
+    </TabItem>
+</Tabs>
+
+* The `start_transaction` call will fail if there is already a transaction in progress.
+
+* During a transaction, the pipeline ingests incoming data without producing output, performing only minimal processing such as resolving primary keys and indexing inputs.
+
+### Commit a transaction
+
+Once the pipeline has ingested all inputs that must be processed as part of the transaction,
+commit the transaction using the [`commit_transaction`](https://docs.feldera.com/api/commit-the-current-transaction) API.
+
+<Tabs>
+    <TabItem value="rest" label="REST API">
+    ```shell
+    $ curl -X POST http://localhost:8080/v0/pipelines/my_pipeline/commit_transaction
+    "Transaction commit initiated"
+    ```
+    </TabItem>
+
+    <TabItem value="python" label="Python SDK">
+    ```python
+    transaction_id = pipeline.commit_transaction(transaction_id, wait = True)
+    ```
+    </TabItem>
+
+    <TabItem value="fda" label="fda CLI">
+    ```shell
+    $ fda commit-transaction my_pipeline
+
+    Transaction committed successfully.
+    ```
+    </TabItem>
+</Tabs>
+
+* The `commit_transaction` REST API initiates the commit. The commit can take some time and complete later.
+  Use the [`/stats` endpoint](#monitoring-transaction-status) to monitor transaction status.
+
+* The Python SDK and `fda` CLI provide both blocking and non-blocking variants of the commit operation.
+  The blocking variant waits until the commit is complete before returning control to the caller.
+
+* During a transaction commit, the pipeline computes updates to all views in the
+  program. Depending on the volume of ingested data and the complexity of the
+  views, this process can take a significant amount of time. While the commit is
+  in progress, the pipeline neither ingests new inputs nor produces outputs. To
+  provide visibility, the pipeline periodically (every 10 seconds) reports the
+  progress of the current commit operation in the log.
+
+* Once the commit is complete, the pipeline outputs a batch of updates for every view in the program.
+  These updates are processed by the output connectors, which send them to their associated data sinks.
+
+When the transaction is complete, the pipeline goes back into continuous processing mode. The user can start a new transaction any time.
+
+### Monitoring transaction status
+
+The user can monitor the transaction handling status of the pipeline using the [`/stats`](https://docs.feldera.com/api/retrieve-statistics-e-g-performance-counters-of-a-running-or-paused-pipeline) endpoint. Possible statuses are summarized
+in the following table:
+
+
+| Status                  | Description                                     |
+|-------------------------|-------------------------------------------------|
+| `NoTransaction`         | There is currently no active transaction.       |
```

Comment:
this does not mean that the pipeline is not doing work - on the contrary, it may be in continuous processing mode.
I wonder whether the state name should reflect that

Reply:
This is not the pipeline's state, this is specifically the current status of transaction processing. I will change the docs to say that it's running in the continuous mode to avoid confusing people into thinking that it's idle in this state.